plugins/inputs/huebridge/huebridge_test.go

```diff
@@ -0,0 +1,95 @@
+package huebridge
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/tdrn-org/go-hue/mock"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestConfig(t *testing.T) {
+	conf := config.NewConfig()
+	err := conf.LoadConfig("testdata/conf/huebridge.conf")
+	require.NoError(t, err)
+	require.Len(t, conf.Inputs, 1)
+	plugin, ok := conf.Inputs[0].Input.(*HueBridge)
+	err = plugin.Init()
+	require.NoError(t, err)
+	require.True(t, ok)
+	require.NotNil(t, plugin.Bridges)
+	require.Len(t, plugin.Bridges, 4)
+	require.NotEmpty(t, plugin.RemoteClientId)
+	require.NotEmpty(t, plugin.RemoteClientSecret)
+	require.NotEmpty(t, plugin.RemoteCallbackUrl)
+	require.NotEmpty(t, plugin.RemoteTokenDir)
+	require.NotNil(t, plugin.RoomAssignments)
+	require.Len(t, plugin.RoomAssignments, 2)
+	require.Equal(t, config.Duration(60*time.Second), plugin.Timeout)
+	require.Equal(t, "secret", plugin.TLSKeyPwd)
+	require.True(t, plugin.InsecureSkipVerify)
+}
```

Comment:
I don't think we need this one. It would be better to test for errors during `Init()` or `Start()` here like unsupported schemes or missing `remote` operations. For this you can directly define the plugin instance like
```
plugin := &HueBridge{
    Bridges: []{"invalid://foo"}
}
```
etc.

Reply:
This actually verifies that the sample.conf structure matches the plugin struct. I have added a test for testing invalid URLs.